### PR TITLE
feat: add COINBASE opcode

### PR DESCRIPTION
**Pull request by alessandromazza98**
Closes https://github.com/lambdaclass/evm_mlir/issues/47

- Implement `COINBASE` opcode
- Add unit tests

### CODE REVIEW

File: tests/evm.rs [hunk 1]
```diff
@@ -454,3 +454,31 @@ fn callvalue_stack_overflow() {
     let env = Env::default();
     run_program_assert_halt(program, env);
 }
+
+#[test]
+fn coinbase_happy_path() {
```
Comment (JulianVentura) [inline tests/evm.rs:459]: Can we add a test (or change this one) to match the example given in [the EVM Codes for COINBASE](https://www.evm.codes/), please?

Comment (alessandromazza98) [reply tests/evm.rs:459]: sure

File: README.md [hunk 2]
```diff
@@ -45,6 +45,7 @@ An EVM-bytecode to machine-bytecode compiler using MLIR and LLVM.
 1. (0x36) CALLDATASIZE
 1. (0x37) CALLDATACOPY
 1. (0x38) CODESIZE
+1. (0x41) COINBASE
```
Comment (MegaRedHand) [inline README.md:48]: Can you put this in order?

**Review by MegaRedHand — APPROVED** [review]
Left a comment about the readme, but should be OK to merge

File: README.md (hunk 2 above)
Comment (alessandromazza98) [reply README.md:48]: yes

File: tests/evm.rs [hunk 3]
```diff
@@ -748,6 +748,39 @@ fn callvalue_stack_overflow() {
...
+        .expect("Decoding failed")
+        .try_into()
+        .expect("Incorrect length");
+    let operations = vec![Operation::Coinbase];
+    let mut env = Env::default();
+    env.block.coinbase = coinbase.into();
+
+    let expected_result = BigUint::from_bytes_be(&coinbase);
+
+    run_program_assert_result(operations, env, expected_result);
```
Comment (MegaRedHand) [inline tests/evm.rs:765]:
```suggestion
    run_program_assert_num_result(operations, env, expected_result);
```

**Comment by MegaRedHand** [comment]
Congrats on your first contribution!

**Pull request merged**